File: Crassus.Native.iOS/WindowDelegate.c
//
//  WindowDelegate.c
//  TheObiG LLC
//
//  Created by SolomonRain on 2/15/16.
//  Copyright © 2016 Crassus. All rights reserved.
//

#include "WindowDelegate.h"

void SetRootViewController(void * window, void * viewController, DoubleObjectAction action)
{
    (*action)(window, viewController);
}

void * GetMasterView(void * window, Exchanger function)
{
    return (*function)(window);
}

File: Crassus.Native.iOS/WindowDelegate.m
//
//  WindowDelegate.m
//  TheObiG LLC
//
//  Created by SolomonRain on 2/15/16.
//  Copyright © 2016 Crassus. All rights reserved.
//

@import UIKit;

void CrassusSetRootViewController(void * window, void * viewController)
{
    ((__bridge UIWindow*)window).rootViewController = CFBridgingRelease(viewController);
}

void * CrassusGetMasterView(void * window)
{
    return (__bridge void *)[((__bridge UIWindow *)window).subviews objectAtIndex:0];
}